nav2_simple_commander/nav2_simple_commander/robot_navigator.py

```diff
@@ -265,11 +265,11 @@ def getResult(self):
         else:
             return TaskResult.UNKNOWN
 
-    def waitUntilNav2Active(self):
+    def waitUntilNav2Active(self, navigator='bt_navigator', localizer='amcl'):
         """Block until the full navigation system is up and running."""
-        self._waitForNodeToActivate('amcl')
+        self._waitForNodeToActivate(navigator)
         self._waitForInitialPose()
```

Comment:
If the localizer is non-AMCL, then it might not make sense to wait for the initial pose (e.g. if in SLAM). I think that check should be done if its AMCL, else to not do it.